### PR TITLE
Re-enable pyupgrade via ruff

### PR DESCRIPTION
**Pull request by dbast**
### Description

#13272 replaced black + pyupgrade via ruff format, but forgot to actually enable the pyupgrade rules for ruff.

This PR enables pyupgrade via ruff again and applies it.

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [ ] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

**Comment by dbast** [comment]
pre-commit.ci autofix

File: conda/common/compat.py [hunk 1]
```diff
@@ -53,14 +54,13 @@ def isiterable(obj):
...
 from collections import OrderedDict as odict  # noqa: F401
-from io import open as io_open  # NOQA
 
 
 def open(
     file, mode="r", buffering=-1, encoding=None, errors=None, newline=None, closefd=True
 ):
     if "b" in mode:
-        return io_open(
+        return builtins.open(
```
[the remark this answers is not in the document]
Comment (dbast) [reply conda/common/compat.py:63]: the function is name open... if you return open then this leads to a non ending recursion :/ ... tried that already before

File: conda/common/compat.py [hunk 2]
```diff
@@ -69,7 +69,7 @@ def open(
             closefd=closefd,
         )
     else:
-        return io_open(
+        return builtins.open(
```
[the remark this answers is not in the document]
Comment (dbast) [reply conda/common/compat.py:72]: see comment above

File: conda/common/compat.py [hunk 3]
```diff
@@ -7,6 +7,7 @@
 # If it's only used in one module, keep it in that module, preferably near the top.
 # This module should contain ONLY stdlib imports.
 
+import builtins
```
[the remark this answers is not in the document]
Comment (dbast) [reply conda/common/compat.py:10]: see comment above

File: conda/common/compat.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (kenodegard) [reply conda/common/compat.py:63]: Ah I missed that, carry on

**Review by travishathaway — APPROVED** [review]

**Pull request merged**